syntax = "proto2";

package mldb;

// Union of all messages from client to server.
message ClientMsg {
  oneof msg {
    ClientHandshake handshake_msg = 1;
    ClientDummyRequest dummy_req = 2;
  }
}

// Union of all messages from server to client.
message ServerMsg {
  oneof msg {
    ServerHandshake handshake_msg = 1;
  }
}

// The first part of server's reply msg contains server router socket's
// identity, so no need to include that.
message ServerHandshake {
  optional int32 client_id = 1;
}

message ClientHandshake {
}

message ClientDummyRequest {
  optional string req = 1;
}
